### DISCUSSION TITLE
clientInstance.getObject Bytestream question

### DISCUSSION THREAD

**Discussion by bmwolf** [awslabs/aws-sdk-swift#1017] [Q&A] [answered]
Hello! I am attempting to get a file from a S3 bucket and write the bytestream locally. I had this working a month or so ago but I guess something got updated in the time being. I am getting an error on line 4 **"Value of type 'ByteStream' has no member 'toBytes'"** I can't find anything in the documentation or anything that changed. If anyone has any insight I would really appreciate it.

Thanks!
```
let output = try await clientInstance.getObject(input: input)
                
1 guard let body = output.body else {
2  return nil
3 }
4 let data = body.toBytes().getData()
5 try data.write(to: fileUrl)
```

**Comment by jbelkins** [comment, marked as answer]
You are correct, the API did change with the introduction of HTTP bidirectional streaming in 0.16.0.

The way to get the data from your S3 object:

```
let output = try await clientInstance.getObject(input: input)
let data: Data?
switch output.body {
case .data(let bodyData):
    data = bodyData
case .stream(let stream):
    data = try stream.readToEnd()
default:
    data = nil
}
```

I recognize that this is a little cumbersome when you just want to get the data and the file is too short to worry about streaming it from S3; I'll look at ways to make this a little more convenient in a future SDK release.

**Reply by bmwolf** [reply]
Awesome, thanks so much! That worked perfectly.

**Comment by jbelkins** [comment]
Issue to track this improvement: https://github.com/awslabs/aws-sdk-swift/issues/1018
